platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/MerkleDbTableConfig.java

```diff
@@ -148,17 +147,24 @@ public MerkleDbTableConfig() {
      *      Hash version
      * @param hashType
      *      Hash type
+     * @param maxNumberOfKeys
+     *      Max number of keys that can be stored in a table.
+     * @param hashesRamToDiskThreshold
+     *      Threshold where we switch from storing internal hashes in ram to storing them on disk.
      */
-    public MerkleDbTableConfig(final short hashVersion, final DigestType hashType) {
+    public MerkleDbTableConfig(
+            final short hashVersion,
+            final DigestType hashType,
+            final long maxNumberOfKeys,
+            final long hashesRamToDiskThreshold) {
         // Mandatory fields
         this.hashVersion = hashVersion;
         this.hashType = hashType;
 
         // Optional hints, may be set explicitly using setters later. Defaults are loaded from
         // MerkleDb configuration
-        final MerkleDbConfig dbConfig = ConfigurationHolder.getConfigData(MerkleDbConfig.class);
-        maxNumberOfKeys = dbConfig.maxNumOfKeys();
-        hashesRamToDiskThreshold = dbConfig.hashesRamToDiskThreshold();
+        this.maxNumberOfKeys = maxNumberOfKeys;
```

Comment:
Does it make sense to check that the provided maxNumberOfKeys (and hashesRamToDiskThreshold) are >= 0 ? A check like this exists in the other constructor

Reply:
I think that is a good idea and copied the checks from the setters: https://github.com/hashgraph/hedera-services/pull/15562/commits/e1fd219ac902a0cc4e45b19bc1b57e7245b63305